frontend/packages/ceph-storage-plugin/src/components/create-storage-system/reducer.ts

```diff
@@ -0,0 +1,86 @@
+import { BackingStorageType, RHCS } from '../../constants/create-storage-system';
+
+export type WizardState = React.ReducerState<WizardReducer>;
+export type WizardDispatch = React.Dispatch<React.ReducerAction<WizardReducer>>;
+
+export type WizardCommonProps = {
+  state: WizardState;
+  dispatch: WizardDispatch;
+};
+
+/* State of CreateStorageSystem */
+export const initialState: CreateStorageSystemState = {
+  currentStep: {
+    count: 1,
+  },
+  backingStorage: {
+    type: BackingStorageType.EXISTING,
+    externalProvider: RHCS,
+  },
+  storageClass: { name: '', provisioner: '' },
+};
+
+type CreateStorageSystemState = {
+  currentStep: StepState;
+  backingStorage: {
+    type: BackingStorageType;
+    externalProvider: string;
```

Comment:
```suggestion
    externalProvider?: string;
```
Optional right ?

Reply:
It's the state of the component. it's set by the external storage dropdown

Reply:
```suggestion
    externalProvider: RHCS | IBM_FLASH;
```
Reduce the scope of values that can be assigned to external providers

Reply:
Expect some of changes in another PR.
Otherwise it will increase the scope. the next PR will standardize contribution mechanism for plugins.